driver/src/models/mod.rs

```diff
@@ -0,0 +1,47 @@
+pub mod state;
```

Comment:
What is with the file name here?  What is mod? There is no struct called mod here...

Reply:
It just says that the module state (code you can find in state.rs) is a public submodule of the models model

Reply:
No, my question is in reference to the file name `mod.rs`

Reply:
`models/mod.rs` contains the code that is exposed by the `models` module (`use crate::models`). Every other file in this folder contains the code exposed by the submodulde (e.g. `models/foo.rs` would be accessed via `use crate::models::foo`)

If there is no other submodule in a folder having `models/mod.rs` or just `models.rs` is equivalent. See more: https://doc.rust-lang.org/rust-by-example/mod/split.html